In [1]:
!pip install ipython-sql

In [2]:
%load_ext sql

In [3]:
%sql mysql://USERNAME:PASSWORD@HOST/DATABASE

'Connected: admin@Robinhood_FinalProject'

Question: What do Robinhood and Morning Brew's articles look like side-by-side on a day to-day-basis? Is there a visual correlation? 

Justification: While this query may seem simple, it is a great query because it gives managers an idea of the general theme from a visual perspective of what both companies write about on a given day. It could also be used to compare how they thematically create their headlines (how do they draw the reader in? Is it funny, serious, ironic?) 

SQL features used: JOIN

In [7]:
%%sql 
SELECT a.`Date`, a.Title, mb.Article_Title
FROM Articles a
JOIN Morning_Brew_Articles mb
ON a.Article_ID = mb.Article_ID;

 * mysql://admin:***@lmu-dev-01.cgixedv0o4jd.us-east-1.rds.amazonaws.com/Robinhood_FinalProject
99 rows affected.


[('Monday, May 3, 2021', "????  Big Tech's big blowout ", 'Buffett Dishes on SPACs, Apple, and more'),
 ('Friday, April 30, 2021', "???? Biden's $1.8T social plan  ", 'Icebreakers with Andrew Ross Sorkin, tackling imposter syndrome, and the debate over politics at work.'),
 ('Thursday, April 29, 2021', '????  Apple and Google really did it', 'AirTags are about a lot more than tracking devices'),
 ('Wednesday, April 28, 2021', "????  Apple's surprising ad play ", 'Things are looking up '),
 ('Tuesday, April 27, 2021', "????  Exxon's Blockbuster moment ", 'Biden Completes the Trilogy '),
 ('Monday, April 26, 2021', '???? 100 days of Biden', 'The Podcast Subscription Wars '),
 ('Friday, April 23, 2021', "?? Southwest's (shocking) profit", 'Tesla had a record quarter, but tensions in China and at home mean more stress for Elon.'),
 ('Thursday, April 22, 2021', '???? Special Earth Day Edition (feat. crypto carbon)', 'Apple is releasing its big privacy update '),
 ('Wednesday, April 21, 2021', " ???? Apple's colorful new gadgets ", 'The $0 movie that started as a joke then became a blockbuster '),
 ('Tuesday, April 20, 2021', '???? A Doge story', 'Simone Biles vaults over to Athleta'),
 ('Monday, April 19, 2021', '???? The future of entertainment', 'A capital gains tax hike could be coming'),
 ('Friday, April 16, 2021', '???? "Revenge Spending" receipts', 'Biden Wants to Ramp Up Vaccine Demand '),
 ('Thursday, April 15, 2021', '????  Big banks, big comeback ', 'Netflix is facing strong headwinds in the new-old normal.'),
 ('Wednesday, April 14, 2021', '????  The largest SPAC merger ever ', 'A record number of Americans support legalizing marijuana.'),
 ('Tuesday, April 13, 2021', '???? The biggest IPO, squashed again', 'Far fewer businesses folded during the pandemic than economists expected.'),
 ('Monday, April 12, 2021', '????  Amazon\'s union "victory" ', 'Can you take time off three months into work? '),
 ('Friday, April 9, 2021', '???? Self-driving IPO ', 'On delis and dogecoin'),
 ('Thursday, April 8, 2021', '???? Coinbase drops big numbers', 'Lumber prices hit an all-time high and Jeff Bezos pens his final shareholder letter.'),
 ('Wednesday, April 7, 2021', "???????? China's digital currency  ", 'Coinbase goes public '),
 ('Tuesday, April 6, 2021', '???????? Corporate America pushes back', "US regulators hit pause on Johnson & Johnson's vaccine."),
 ('Monday, April 5, 2021', "???? Biden's $2T makeover plan ", 'Microsoft is making its second-largest acquisition in history'),
 ('Wednesday, March 31, 2021', "????  Bitcoin's rice cooker moment ", 'Chips wanted'),
 ('Tuesday, March 30, 2021', "?? Big banks' big margin call", 'The big ballot showdown at the Arctic Circle '),
 ('Monday, March 29, 2021', '????  Snacks Special Edition: End-of-Quarter Review', 'Rich New Yorkers Will Face Highest Income Tax in Nation '),
 ('Friday, March 26, 2021', "????  You're gonna need a smaller boat", "The ballots are counted for Amazon's historic union vote"),
 ('Thursday, March 25, 2021', '???? Taco Bell goes full iPhone', 'The fight over the corporate tax hike'),
 ('Wednesday, March 24, 2021', '????  GameStop drops the numbers ', 'US Employers Scrambling to Find Workers '),
 ('Tuesday, March 23, 2021', '????  Zoom-as-a-Service', 'LG exits the smartphone arena and Google wins a big victory over Oracle'),
 ('Monday, March 22, 2021', '????  NFTs = crypto Sharpie ', 'Markets revert to the meme '),
 ('Friday, March 19, 2021', ' ????  Williams-Sonoma has you for life', 'Good morning, Happy Easter, and welcome to Sunday Edition! '),
 ('Thursday, March 18, 2021', "???????? Uber's big gig loss ", 'March Jobs Report = Yuge Success '),
 ('Wednesday, March 17, 2021', '???? Operation Overtake Tesla', 'Companies are firming up plans to return to the office'),
 ('Tuesday, March 16, 2021', "????  America's most valuable unicorn ", 'Biden Goes Big on Infrastructure '),
 ('Monday, March 15, 2021', '???? The stimulus has landed ', 'The creator economy is hitting 

Morning Brew's titles seem to be shorter and to the point, while also employign more playful language. Furthermore, it seems that on most days, both publications seem to have different topics for their headlines. We reccommend taking a deeper look at how cusotmers respond to the headline conventions

Purpose of Query: to select the FAANG companies from the fortune 500 database for the purpose of determining their ranks and revenues, to be used in further queries, and to group those into a CTE for easy reference since these are companies that are considered to be extremely relevant and therefore we want to assess how well Robinhood and Morning brew are covering each stock

Features Used: CTE, GROUP BY 

In [8]:
%%sql 
WITH FAANG_companies_in_fortune_500_CTE AS (
	SELECT * 
	FROM Fortune_500_Data 
	WHERE Company_Name IN ('Facebook', 'Amazon.com', 'Apple', 'Netflix', 'Alphabet')
	)
SELECT 
	Company_Name, 
	Company_Rank,
	Revenues as Revenue
FROM FAANG_companies_in_fortune_500_CTE
GROUP BY Company_Name; 

 * mysql://admin:***@lmu-dev-01.cgixedv0o4jd.us-east-1.rds.amazonaws.com/Robinhood_FinalProject
5 rows affected.


[('Amazon.com', 2, '$280,522'),
 ('Apple', 4, '$260,174'),
 ('Alphabet', 11, '$161,857'),
 ('Facebook', 46, '$70,697'),
 ('Netflix', 164, '$20,156')]

As can be seen, the FAANG companies are all in the fortune 500 list, 4 of which are in the top 50. We also can see the revenues, which determine rank as we've now found out

This query creates a VIEW for all of the morning brew article that mention any of the FAANG companies. I chose a VIEW because it packages all the FAANG articles published by morning brew in an easily accessible format and makes queries based on that gathered lot of information easier for analysis

FEATURES USED: VIEW

In [9]:
%%sql CREATE OR REPLACE VIEW mb_faang_articles AS 
SELECT 
	Article_ID,
	`Date`,
	Article_Title 
FROM Morning_Brew_Articles 
WHERE 
	Article_Title LIKE '%Facebook%' OR 
	Article_Title LIKE '%Amazon%' OR 
	Article_Title LIKE '%Apple%' OR 
	Article_Title LIKE '%Netflix%' OR 
	Article_Title LIKE '%Google%'; 

 * mysql://admin:***@lmu-dev-01.cgixedv0o4jd.us-east-1.rds.amazonaws.com/Robinhood_FinalProject
0 rows affected.


[]

Now, a VIEW has been created (we haven't actually discovered anything yet but it will be useful down the road)

This query creates a VIEW for all of the Robinhood articles that mention any of the FAANG companies. I chose a VIEW because it packages all the FAANG articles published by Robinhood in an easily accessible format and makes queries based on that gathered lot of information easier for analysis

In [11]:
%%sql 
CREATE OR REPLACE VIEW robinhood_faang_articles AS
SELECT
Article_ID,
`Date`,
Title
FROM Articles 
WHERE 
	Title LIKE '%Facebook%' OR 
	Title LIKE '%Amazon%' OR 
	Title LIKE '%Apple%' OR 
	Title LIKE '%Netflix%' OR 
	Title LIKE '%Google%';

 * mysql://admin:***@lmu-dev-01.cgixedv0o4jd.us-east-1.rds.amazonaws.com/Robinhood_FinalProject
0 rows affected.


[]

Now, a VIEW has been created (we haven't actually discovered anything yet but it will be useful down the road)

Question: Which articles, if any, published about the FAANG companies, were published on the same day? (Notice, date isn't used here but the article IDs correspond to the same dates across the two tables. for example, monday may 3 has the same article ID (1), in both the Robinhood and Morning Brew tables. 

Business Justification: this allows managers from morning brew or robinhood to gauge how well and consistently they are aligning with the competitor's coverage of FAANG companies on a daily basis and can inform them of where one may be pulling ahead with overall coverage

SQL used: CASE, JOIN

In [12]:
%%sql
SELECT 
	r.Title, 
	mb.Article_Title,
	r.Article_ID,
	CASE 
		WHEN mb.Article_Title IS NULL THEN 'No Match'
			ELSE 'Match'
	END AS MatchingContentCheck
FROM robinhood_faang_articles r
LEFT JOIN mb_faang_articles mb
ON r.Article_ID = mb.Article_ID;

 * mysql://admin:***@lmu-dev-01.cgixedv0o4jd.us-east-1.rds.amazonaws.com/Robinhood_FinalProject
15 rows affected.


[('????  Apple and Google really did it', None, 3, 'No Match'),
 ("????  Apple's surprising ad play ", None, 4, 'No Match'),
 (" ???? Apple's colorful new gadgets ", None, 9, 'No Match'),
 ('????  Amazon\'s union "victory" ', None, 16, 'No Match'),
 ("???? Amazon's union nightmare ", None, 37, 'No Match'),
 ('???? Discord = the anti-Facebook', None, 38, 'No Match'),
 ("????????  Google's cookies crumble ", None, 41, 'No Match'),
 ("???? Target's anti-Amazonian strategy ", None, 42, 'No Match'),
 ('???? Walmart, Amazon, and the minimum wage ', None, 49, 'No Match'),
 ('????  Facebook says "no news, friends"', None, 50, 'No Match'),
 ('???? Ad Bowl: Snap vs. Facebook', 'Google Keeps Its Hand Out of the Cookie Jar ', 58, 'Match'),
 ('????  Walmart, Amazon, and the vaccine rollout ', None, 68, 'No Match'),
 ("????  Amazon's anti-return policy ", None, 76, 'No Match'),
 ("?? Apple's iCar is alive ", None, 89, 'No Match'),
 ("???? Facebook's family breakup ", None, 97, 'No Match')]

As can be seen, only one match occurs on the same day between the two companies, and ironically they aren't even on the same company, just both FAANG stocks. Clearly, we see little correlation between FAANG coverage on a per-day basis between the two companies. Thus, it would appear Morning Brew is showing better coverage of FAANG and Robinhood is not matching that coverage

The purpose of this query was to explore how dates and coverage of another topic were related, in this case coverage of President Biden. 

In [13]:
%%sql
SELECT Title, `Date` 
FROM Articles
WHERE Title LIKE '%Biden%'
UNION 
SELECT Article_Title, `Date` 
FROM Morning_Brew_Articles 
WHERE Article_Title Like '%Biden%'; 

 * mysql://admin:***@lmu-dev-01.cgixedv0o4jd.us-east-1.rds.amazonaws.com/Robinhood_FinalProject
13 rows affected.


[("???? Biden's $1.8T social plan  ", 'Friday, April 30, 2021'),
 ('???? 100 days of Biden', 'Monday, April 26, 2021'),
 ("???? Biden's $2T makeover plan ", 'Monday, April 5, 2021'),
 ('???? Biden cooks up a plan ', 'Friday, February 26, 2021'),
 ('Biden Completes the Trilogy ', 'Apr 29, 2021'),
 ('Biden Wants to Ramp Up Vaccine Demand ', 'Apr 22, 2021'),
 ('Biden Goes Big on Infrastructure ', 'Apr 1, 2021'),
 ("Previewing Biden's Tax Plan ", 'Mar 16, 2021'),
 ('President Biden signed his first major bill: a $1.9T relief package', 'Mar 12, 2021'),
 ("The Senate Passes Biden's $1.9 Trillion Stimulus Bill ", 'Mar 8, 2021'),
 ("Biden's packed first day ", 'Jan 21, 2021'),
 ('Will the Keystone XL survive the Biden administration? ', 'Jan 19, 2021'),
 ('Biden Unveils His Economic Plan', 'Jan 15, 2021')]

There appears to be no matching dates for articles published on Biden from either publisher. Morning Brew appears to cover biden more.

This query was used to get the total number of employees, and the average employee count by company

SQL feature used: WINDOW FUNCTION

In [24]:
%%sql 
SELECT 
	SUM(Number_of_Employees),
	AVG(Number_of_Employees) OVER 
		(Partition by Company_Name) AS avg_num_employees_by_company
FROM Fortune_500_Data; 

 * mysql://admin:***@lmu-dev-01.cgixedv0o4jd.us-east-1.rds.amazonaws.com/Robinhood_FinalProject
1 rows affected.


[(Decimal('29192045'), Decimal('2200000.0000'))]

Discovery: the total number of employees in the fortune 500 dataset is 29192045, whereas the average count is 22000. 

This query gives the top 100 companies from the fortune 500 dataset

In [20]:
%%sql 
SELECT *
FROM Fortune_500_Data
WHERE company_rank <= 100; 

 * mysql://admin:***@lmu-dev-01.cgixedv0o4jd.us-east-1.rds.amazonaws.com/Robinhood_FinalProject
100 rows affected.


[(1, 1, 'Walmart', 2200000, '$523,964', '$321,803'),
 (2, 2, 'Amazon.com', 798000, '$280,522', '$970,680'),
 (3, 3, 'Exxon Mobil', 74900, '$264,938', '$160,696'),
 (4, 4, 'Apple', 137000, '$260,174', '$1,112,641'),
 (5, 5, 'CVS Health', 290000, '$256,776', '$77,376'),
 (6, 6, 'Berkshire Hathaway', 391500, '$254,616', '$442,897'),
 (7, 7, 'UnitedHealth Group', 325000, '$242,155', '$236,555'),
 (8, 8, 'McKesson', 70000, '$214,319', '$21,845'),
 (9, 9, 'AT&T', 247800, '$181,193', '$209,388'),
 (10, 10, 'AmerisourceBergen', 21500, '$179,589', '$18,221'),
 (11, 11, 'Alphabet', 118899, '$161,857', '$798,905'),
 (12, 12, 'Ford Motor', 190000, '$155,900', '$19,151'),
 (13, 13, 'Cigna', 73700, '$153,566', '$65,897'),
 (14, 14, 'Costco Wholesale', 201500, '$152,703', '$125,908'),
 (15, 15, 'Chevron', 48200, '$146,516', '$136,176'),
 (16, 16, 'Cardinal Health', 49500, '$145,534', '$13,988'),
 (17, 17, 'JPMorgan Chase', 256981, '$142,422', '$276,750'),
 (18, 18, 'General Motors', 164000, '$137,237', '$29,695'),
 (19, 19, 'Walgreens Boots Alliance', 287000, '$136,866', '$40,528'),
 (20, 20, 'Verizon Communications', 135000, '$131,868', '$222,220'),
 (21, 21, 'Microsoft', 144000, '$125,843', '$1,199,550'),
 (22, 22, 'Marathon Petroleum', 60910, '$124,813', '$15,353'),
 (23, 23, 'Kroger', 435000, '$122,286', '$24,114'),
 (24, 24, 'Fannie Mae', 7500, '$120,304', '$1,841'),
 (25, 25, 'Bank of America', 208131, '$113,589', '$185,227'),
 (26, 26, 'Home Depot', 415700, '$110,225', '$200,665'),
 (27, 27, 'Phillips 66', 14500, '$109,559', '$23,490'),
 (28, 28, 'Comcast', 190000, '$108,942', '$156,533'),
 (29, 29, 'Anthem', 70600, '$104,213', '$57,245'),
 (30, 30, 'Wells Fargo', 259800, '$103,915', '$117,366'),
 (31, 31, 'Citigroup', 200000, '$103,449', '$88,377'),
 (32, 32, 'Valero Energy', 10222, '$102,729', '$18,532'),
 (33, 33, 'General Electric', 205000, '$95,214', '$69,406'),
 (34, 34, 'Dell Technologies', 165000, '$92,154', '$29,246'),
 (35, 35, 'Johnson & Johnson', 132200, '$82,059', '$345,705'),
 (36, 36, 'State Farm Insurance', 57672, '$79,395', '-'),
 (37, 37, 'Target', 368000, '$78,112', '$46,574'),
 (38, 38, 'IBM', 383800, '$77,147', '$98,551'),
 (39, 39, 'Raytheon Technologies', 243200, '$77,046', '$81,705'),
 (40, 40, 'Boeing', 161100, '$76,559', '$84,149'),
 (41, 41, 'Freddie Mac', 6892, '$75,125', '$909'),
 (42, 42, 'Centene', 56600, '$74,639', '$34,862'),
 (43, 43, 'United Parcel Service', 377640, '$74,094', '$80,196'),
 (44, 44, "Lowe's", 260000, '$72,148', '$64,963'),
 (45, 45, 'Intel', 110800, '$71,965', '$231,662'),
 (46, 46, 'Facebook', 44942, '$70,697', '$475,455'),
 (47, 47, 'FedEx', 389500, '$69,693', '$31,679'),
 (48, 48, 'MetLife', 49000, '$69,620', '$27,997'),
 (49, 49, 'Walt Disney', 223000, '$69,570', '$174,405'),
 (50, 50, 'Procter & Gamble', 97000, '$67,684', '$271,640'),
 (51, 51, 'PepsiCo', 267000, '$67,161', '$166,848'),
 (52, 52, 'Humana', 46000, '$64,888', '$41,490'),
 (53, 53, 'Prudential Financial', 51511, '$64,807', '$20,650'),
 (54, 54, 'Archer Daniels Midland', 38100, '$64,656', '$19,603'),
 (55, 55, 'Albertsons', 267000, '$60,535', '-'),
 (56, 56, 'Sysco', 69000, '$60,114', '$23,203'),
 (57, 57, 'Lockheed Martin', 110000, '$59,812', '$95,539'),
 (58, 58, 'HP', 56000, '$58,756', '$24,821'),
 (59, 59, 'Energy Transfer', 12812, '$54,213', '$12,374'),
 (60, 60, 'Goldman Sachs Group', 38300, '$53,922', '$55,417'),
 (61, 61, 'Morgan Stanley', 60431, '$53,823', '$52,102'),
 (62, 62, 'Caterpillar', 102300, '$53,800', '$63,832'),
 (63, 63, 'Cisco Systems', 75900, '$51,904', '$166,709'),
 (64, 64, 'Pfizer', 88300, '$51,750', '$181,075'),
 (65, 65, 'HCA Healthcare', 245000, '$51,336', '$30,411'),
 (66, 66, 'AIG', 46000, '$49,746', '$20,886'),
 (67, 67, 'American Express', 64500, '$47,020', '$68,983'),
 (68, 68, 'Delta Air Lines', 91224, '$47,007', '$18,262'),
 (69, 69, 'Merck', 71000, '$46,840', '$195,141'),
 (70, 70, 'American Airlines Group', 133700, '$45,768', '$5,194'),
 (71, 71, 'Charter Communications', 95100, 

This gives exactly what the query asked for: the top 100 companies including their name, number of employees, revenues, and market value

In [25]:
!pip install nbmerge

  Created wheel for nbmerge: filename=nbmerge-0.0.4-py2.py3-none-any.whl size=6406 sha256=bf69bf84c74179d3efe30946eb71ba72d455fd33ea8977e80b6da71824f0b9d5
  Stored in directory: /Users/billis/Library/Caches/pip/wheels/2d/3c/a9/8753a8ef327aca9c908d4f150628b87b548e1659ecc5fe1b54
Successfully built nbmerge


In [1]:
!nbmerge data_collection.ipynb sql_analysis.ipynb > presentation.ipynb